### PR TITLE
11.0 mig base country state translatable

### PR DESCRIPTION
**Pull request by yostashiro**

### CODE REVIEW

**Comment by yostashiro** [comment]
CI builds seem to be failing due to this module https://github.com/OCA/partner-contact/tree/11.0/partner_contact_in_several_companies?

File: base_country_state_translatable/README.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline base_country_state_translatable/README.rst]: Please use newest template with png file

File: base_country_state_translatable/README.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+========================
+Translate Country States
+========================
+
+This module allows to translate the country states names
```
Comment (elicoidal) [inline base_country_state_translatable/README.rst]: add a period

File: base_country_state_translatable/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline base_country_state_translatable/__init__.py]: in v11, coding line is not necessary.
In general in `__init__.py`, no need copyright/license

File: base_country_state_translatable/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+{
+    "name": "Translate Country States",
+    "version": "11.0.1.0.0",
+    "depends": ['base',
+                ],
+    'application': False,
+    'author': 'initOS GmbH, Odoo Community Association (OCA)',
+    "category": "Localisation/Europe",
+    'license': 'AGPL-3',
+    'data': [],
```
Comment (elicoidal) [inline base_country_state_translatable/__manifest__.py]: remove empty keys

**Review by elicoidal — APPROVED** [review]
LGTM (some nitpicking)
2 questions (nice to have):
* Do we need migration to create the translation for existing data?
* UT could be added although not very useful IMO

**Comment by yostashiro** [comment]
Thanks @elicoidal for your comments.

> * Do we need migration to create the translation for existing data?

I wasn't sure what to do with the existing translation.  Should the .po files be removed?

**Comment by elicoidal** [comment]
Actually this is not my question. I am not sure of the standard Odoo behavior in case you add the module with existing data: will it create the translation lines with a copy of the name as translated value?

It is more out of curiosity: other than LGTM
@yostashiro

**Review by bealdav — APPROVED** [review]
LGTM, only 3 lines of code here.
No needs UT.
Thanks

**Comment by yostashiro** [comment]
@elicoidal 

> will it create the translation lines with a copy of the name as translated value?

Yes, when you do Translations > Application Terms > Generate Missing Terms.

**Comment by elicoidal** [comment]
> Yes, when you do Translations > Application Terms > Generate Missing Terms.

This is my point: you might need a post_hook trigger to launch the procedure. As said for me nice to have so we can move as is
@yostashiro

**Comment by yostashiro** [comment]
@elicoidal 

> you might need a post_hook trigger to launch the procedure.

In my opinion we should not do this.  Odoo by default does not generate translation lines upon module installation for the terms that are missing translation, so we may just go with this behavior.  Besides the usage of this module is to "Inherit this module in your l10n_xx_country_states Module and translate the states names" as the readme says. ;)

**Review by TimLai125 — APPROVED** [review]

**Review by yajo — CHANGES_REQUESTED** [review]
Please check if your addon is somehow introducing an incompatibility to give those ❌ in bots. I'm afraid that current 11.0 branch is 💚, and I wouldn't want to see it go ❌ when merging this.

Other than that, code looks OK, thanks!

**Comment by yostashiro** [comment]
Sorry about my late follow-up.  I pushed an empty commit to run CI builds again, yet runbot fails with following error which comes from https://github.com/OCA/partner-contact/tree/11.0/partner_contact_in_several_companies:

```
2018-02-10 10:08:51,184 151 INFO openerp_test odoo.addons.base.ir.ir_ui_view: Field 'contact_id' used in attributes must be present in view but is missing:
- 'contact_id' in attrs="{'invisible': ['|',('is_company','=',True),('contact_id','!=',False)]}"

Error context:
View `res.partner.form.contact`
[view_id: 243, xml_id: n/a, model: res.partner, parent_id: 112]
2018-02-10 10:08:51,186 151 WARNING openerp_test odoo.modules.loading: Transient module states were reset
2018-02-10 10:08:51,186 151 ERROR openerp_test odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
File "/home/odoo/OCB-11.0/odoo/models.py", line 967, in _validate_fields
check(self)
File "/home/odoo/OCB-11.0/odoo/addons/base/ir/ir_ui_view.py", line 338, in _check_xml
self.postprocess_and_fields(view.model, view_doc, view.id)
File "/home/odoo/OCB-11.0/odoo/addons/base/ir/ir_ui_view.py", line 1024, in postprocess_and_fields
self.raise_view_error("\n".join(msg_lines), view_id)
File "/home/odoo/OCB-11.0/odoo/addons/base/ir/ir_ui_view.py", line 506, in raise_view_error
raise ValueError(message)
ValueError: Field 'contact_id' used in attributes must be present in view but is missing:
- 'contact_id' in attrs="{'invisible': ['|',('is_company','=',True),('contact_id','!=',False)]}"

Error context:
View `res.partner.form.contact`
[view_id: 243, xml_id: n/a, model: res.partner, parent_id: 112]
```

https://github.com/OCA/partner-contact/blob/4cf518933989e474c0df8add4341fbe99baf9e37/partner_contact_in_several_companies/views/res_partner.xml#L52

I am not sure why this error is occurring since `contact_id` is present in the view.

@Yajo How do we proceed from here?

**Comment by pedrobaeza** [comment]
Have you rebased on latest main branch?

**Comment by yostashiro** [comment]
I just rebased and yes, that was it.  Thank you @pedrobaeza !

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Could you please squash all migration-related commits together to make the merge?

**Comment by yostashiro** [comment]
@Yajo Thanks for your advice.  Just squashed commits.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**